crates/wdk-alloc/src/lib.rs

```diff
@@ -45,33 +47,65 @@ mod kernel_mode {
     /// <= `DISPATCH_LEVEL`
     pub struct WdkAllocator;
 
-    // The value of memory tags are stored in little-endian order, so it is
-    // convenient to reverse the order for readability in tooling (ie. Windbg)
-    const RUST_TAG: ULONG = u32::from_ne_bytes(*b"rust");
+    /// The value of memory tags are stored in little-endian order, so it is
+    /// convenient to reverse the order for readability in tooling (ie. Windbg).
+    /// The default tag value is `rust`. You may use `store` method to mutate its value.
+    /// However, it's strongly recommended that you mutate the tag for only once.
+    pub static GLOBAL_POOL_TAG: AtomicU32 = AtomicU32::new(u32::from_ne_bytes(*b"rust"));
+    
+    // The minimum alignment of pointer returned by ExAllocatePool2.
+    const POOL_ALIGNMENT: usize = size_of::<*mut u8>() * 2;
 
     // SAFETY: This is safe because the Wdk allocator:
     //         1. can never unwind since it can never panic
     //         2. has implementations of alloc and dealloc that maintain layout
-    //            constraints (FIXME: Alignment of the layout is currenty not
-    //            supported)
+    //            constraints (including size and alignment)
     unsafe impl GlobalAlloc for WdkAllocator {
         unsafe fn alloc(&self, layout: Layout) -> *mut u8 {
             let ptr =
                 // SAFETY: `ExAllocatePool2` is safe to call from any `IRQL` <= `DISPATCH_LEVEL` since its allocating from `POOL_FLAG_NON_PAGED`
                 unsafe {
```

Comment:
restructure the logic so that you only have one unsafe op per unsafe block. this may also help with overuse of nesting conditionals